### PR TITLE
Initial request 

### PR DESCRIPTION
**Pull request by arun-121**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
```
Comment (Febinvarghese2) [inline public/index.html]: @arun-121 can you remove this commented code

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (Febinvarghese2) [inline public/index.html]: @arun-121 can you change the title of the app

File: public/logo192.png
Comment (Febinvarghese2) [inline public/logo192.png:1]: @arun-121 is this icons are needed

File: public/robots.txt
Comment (Febinvarghese2) [inline public/robots.txt:1]: @arun-121 robots.txt can be removed, since we are not avoiding engine crawlers

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  setSortedFilter,
+  setTheme,
+} from "./redux/actions";
+import { Flex, Select, Text, Button, toast } from "@sparrowengg/twigs-react";
+
+function App() {
+  const dispatch = useDispatch();
+  return (
+    <>
+      <Text css={{ backgroundColor: "$primary" }}>this is a sample text</Text>
```
Comment (Febinvarghese2) [inline src/App.js]: @arun-121 is this file needed???

Comment (Febinvarghese2) [reply src/App.js]: @arun-121 why do we have only one component here

File: src/App.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
Comment (Febinvarghese2) [inline src/App.test.js]: @arun-121 we are not testing this app right?, we can remove this unused test file

File: src/components/Bag.jsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import { useDispatch, useSelector } from "react-redux";
+import { Flex, Popover, Text, Button } from "@sparrowengg/twigs-react";
+import { removeFromCart, setTotalAmount } from "../redux/actions";
+import { useNavigate } from "react-router-dom";
+import { useEffect, useMemo } from "react";
+const Bag = () => {
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 can we have a better component name?

File: src/components/Bag.jsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+      tot = tot + Number(ele.cartCount) * Number(ele.price);
+    });
+    tot = Number(tot.toFixed(3));
+    dispatch(setTotalAmount(tot));
+  }, [cart]);
+
+  // const bagTotal = useMemo(() => {
+  //   let tot = 0;
+  //   cart.forEach((ele) => {
+  //     tot = tot + Number(ele.cartCount) * Number(ele.price);
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 can you remove this commented code, can you check the entire PR

File: src/components/Bag.jsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+import { useNavigate } from "react-router-dom";
+import { useEffect, useMemo } from "react";
+const Bag = () => {
+  const cart = useSelector((state) => state.cart);
+  const totalAmount = useSelector((state) => state.totalAmount);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    let tot = 0;
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 can you give meaningful names for the variables

File: src/components/Bag.jsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  //     tot = tot + Number(ele.cartCount) * Number(ele.price);
+  //   });
+  //   return tot.toFixed(2);
+  // }, [cart]);
+  return (
+    <>
+      <Flex
+        flexDirection="column"
+        justifyContent="space-between"
+        gap="10px"
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 we have our theming values for example 10px is considered as $5, can you check the entire pr and here is the documentation for reference https://twigs.surveysparrow.com/docs/theming

Comment (Febinvarghese2) [reply src/components/Bag.jsx]: @arun-121 spaces like margin, gap and padding ranges from $1 to $50, which each values is multiples of two for example $4 -> 8px,

for height and width it is multiples of four for height and width -> $4 -> 16px

File: src/components/Bag.jsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        css={{ width: "300px", overflow: "scroll", height: "100%" }}
+      >
+        <Text
+          css={{
+            marginTop: "30px",
+            marginBottom: "30px",
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 this can be marginBlock: "$15"

File: src/components/Bag.jsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+            height: "50px",
+            textAlign: "center",
+            fontSize: "2rem",
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 instead of using fontSizes here, can you use fontsizes or even Heading component ?

File: src/components/Bag.jsx [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+                }
+                width={"80px"}
+                height={"80px"}
+                alt=""
+              />
+            </Flex>
+          </Flex>
+        )}
+        <Text weight={"bold"} css={{ textAlign: "center" }}>
+          Bag Total : ₹ {totalAmount}
```
Comment (Febinvarghese2) [inline src/components/Bag.jsx]: @arun-121 this should be a template literal right?

**Review by Febinvarghese2 — CHANGES_REQUESTED** [review]
please check these comments

File: public/index.html (hunk 1 above)
Comment (arun-121) [reply public/index.html]: removed the commented code

File: public/index.html (hunk 2 above)
Comment (arun-121) [reply public/index.html]: changed the title as E-Commerce

File: public/logo192.png
Comment (arun-121) [reply public/logo192.png:1]: removed icons

File: src/App.js (hunk 3 above)
Comment (arun-121) [reply src/App.js]: removed the same

File: src/App.test.js (hunk 4 above)
Comment (arun-121) [reply src/App.test.js]: removed the file

File: src/components/Bag.jsx (hunk 5 above)
Comment (arun-121) [reply src/components/Bag.jsx]: changed from Bag to BagItems

File: src/components/Bag.jsx (hunk 7 above)
Comment (arun-121) [reply src/components/Bag.jsx]: changed from tot to total

File: src/components/Bag.jsx (hunk 10 above)
Comment (arun-121) [reply src/components/Bag.jsx]: used Heading component ,removed fontSize property

File: src/components/Bag.jsx (hunk 11 above)
Comment (arun-121) [reply src/components/Bag.jsx]: made it as template literal

File: src/components/Bag.jsx (hunk 6 above)
Comment (arun-121) [reply src/components/Bag.jsx]: removed the same

File: public/robots.txt
Comment (arun-121) [reply public/robots.txt:1]: removed the same

File: src/components/Bag.jsx (hunk 8 above)
Comment (arun-121) [reply src/components/Bag.jsx]: used theming variables

File: src/components/Bag.jsx (hunk 9 above)
Comment (arun-121) [reply src/components/Bag.jsx]: changed as mentioned

File: src/components/BagItems.jsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+          css={{
+            marginBlock: "$15",
+            textAlign: "center",
+          }}
+        >
+          My Bag
+        </Heading>
+
+        <Flex
+          css={{ flexWrap: "wrap", overflow: "scroll" }}
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 we have wrap prop for the flex component

Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 by default if we set to scroll means the scrollbar would be there right?

File: src/components/BagItems.jsx [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+        >
+          {cart.map((ele) => (
+            <Flex
+              justifyContent="center"
+              alignItems="center"
+              key={ele.id}
+              css={{
+                width: "90px",
+                height: "90px",
+                background: "#ffffff",
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 we can use $white900

File: src/components/BagItems.jsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+                borderRadius: "100%",
+                position: "relative",
+              }}
+            >
+              <img src={ele.image} alt="" width={"40px"} height={"40px"} />
+              <Flex
+                css={{
+                  position: "absolute",
+                  width: "20px",
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 this can be $5

File: src/components/BagItems.jsx [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+                  height: "20px",
+                  borderRadius: "100%",
+                  top: 10,
+                  left: 0,
+                  background: "red",
+                  color: "#ffffff",
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 can you checkout our theming values?

File: src/components/BagItems.jsx [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+                }}
+              >
+                <Text css={{ textAlign: "center", width: "100%" }}>
+                  {ele.cartCount}
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 instead of ele can you provide a relevant name??

File: src/components/BagItems.jsx [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+                </Text>
+              </Flex>
+            </Flex>
+          ))}
+        </Flex>
+        {cart.length == 0 && (
```
Comment (Febinvarghese2) [inline src/components/BagItems.jsx]: @arun-121 this can be `!cart.length`

File: src/components/CartItems.jsx [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+import EmptyCartUI from "./EmptyCart";
+const CartItems = () => {
+  const cart = useSelector((state) => state.cart);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  return (
+    <>
+      {cart.length === 0 ? (
+        // <Flex css={{ width: "100%" }}>
```
Comment (Febinvarghese2) [inline src/components/CartItems.jsx]: @arun-121 can you remove this commented code, can you check for the entire PR

File: src/components/CartItems.jsx [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+        <EmptyCartUI />
+      ) : (
+        // </Flex>
+        <Flex flexDirection="column" css={{ overflow: "scroll" }}>
+          <Text
+            css={{
+              marginBlock: "$15",
+              textAlign: "center",
+              fontSize: "$3xl",
```
Comment (Febinvarghese2) [inline src/components/CartItems.jsx]: @arun-121 can you use Heading component instead?

File: src/components/CartItems.jsx [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+          <Flex
+            flexDirection="column"
+            css={{ overflow: "scroll", gap: "$15", padding: "$10" }}
+          >
+            {cart.map((ele) => (
+              <Flex
+                key={ele.id}
+                css={{
+                  background: "#ffffff",
+                  borderRadius: "20px",
```
Comment (Febinvarghese2) [inline src/components/CartItems.jsx:37]: @arun-121 can you checkout our theming values

File: src/components/CartItems.jsx [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+                    {ele.title}
+                  </Text>
+                  <Text css={{ color: "#60695C" }}>{ele.category}</Text>
+                  <Text css={{ color: "$primary" }}>{ele.description}</Text>
+                  <Flex alignItems="center" gap="$5">
+                    <ReactStarsRating isEdit={false} value={ele.rating.rate} />
+                    <Text weight={"bold"}>{ele.rating.rate}/5</Text>
+                  </Flex>
+                  <Flex justifyContent="space-between">
+                    <Text weight={"bold"}>₹ {ele.price}</Text>
```
Comment (Febinvarghese2) [inline src/components/CartItems.jsx]: @arun-121 can you wrap this in a template literals??

File: src/components/CartItems.jsx [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+                        onClick={() => {
+                          dispatch(removeFromCart(ele.id));
+                        }}
+                        css={{ color: "red", cursor: "pointer" }}
+                        weight={"bold"}
+                      >
+                        -
+                      </Text>
+                      <Text weight={"bold"}>{ele.cartCount}</Text>
+                      <Text
```
Comment (Febinvarghese2) [inline src/components/CartItems.jsx]: @arun-121 instead of Text component can you use the button component

File: src/components/CheckoutDialog.jsx [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+  DialogTrigger,
+  DialogContent,
+  Button,
+  Flex,
+} from "@sparrowengg/twigs-react";
+import ShippingAddress from "./ShippingAddress";
+import OrderSummary from "./OrderSummary";
+export const CheckoutDialog = () => {
+  return (
+    <>
```
Comment (Febinvarghese2) [inline src/components/CheckoutDialog.jsx]: @arun-121 is this empty fragment needed???

File: src/components/CheckoutDialog.jsx [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+          <Button size="sm" css={{ padding: "$10", marginBottom: "$5" }}>
+            Click to Proceed
+          </Button>
+        </DialogTrigger>
+        <DialogContent>
+          <DialogTitle></DialogTitle>
+          <ShippingAddress />
+          <OrderSummary />
+
+          <Flex>
```
Comment (Febinvarghese2) [inline src/components/CheckoutDialog.jsx]: @arun-121 is this flex component needed???

File: src/components/EmptyCart.jsx [hunk 25]
```diff
@@ -0,0 +1,31 @@
+import { useNavigate } from "react-router-dom";
+import { Flex, Heading, Text, Button } from "@sparrowengg/twigs-react";
+const EmptyCart = () => {
```
Comment (Febinvarghese2) [inline src/components/EmptyCart.jsx:4]: @arun-121 can you add a space between imports and function declaration.

File: src/components/EmptyCart.jsx [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+      <Flex
+        flexDirection="column"
+        justifyContent="space-evenly"
+        alignItems="center"
+        css={{ height: "100%" }}
+      >
+        <Heading css={{ textAlign: "center" }}>
+          Your cart is currently empty!
+        </Heading>
+        <img src={"./empty-cart.png"} alt="" />
```
Comment (Febinvarghese2) [inline src/components/EmptyCart.jsx]: @arun-121 can you try to add alt text what if the img breaks, it may show the text right?

File: src/components/FilterMenu.jsx [hunk 27]
```diff
@@ -0,0 +1,135 @@
...
+                  <Checkbox
+                    id="100rs"
+                    size="md"
+                    onChange={(e) => dispatch(setPriceFilter(100))}
+                  >
+                    <Text>under 100</Text>
+                  </Checkbox>
+                  <Checkbox
+                    onChange={(e) => dispatch(setPriceFilter(200))}
+                    id="200rs"
```
Comment (Febinvarghese2) [inline src/components/FilterMenu.jsx]: @arun-121 this can be just 200 right?, and can you try to move these to a separate constants file,
so what are the contents we need to put in constants : hard coded strings in the code can be moved to constants since the same values can be used in other places of the code.
if you want to change an constant means you don't have to change the entire files which you had used, instead you can change it in the constants itself

File: src/components/HomeSkeleton.jsx [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+      justifyContent="center"
+      css={{
+        flexWrap: "wrap",
+        gap: "$10",
+        overflow: "scroll",
+        padding: "$2",
+        width: "100%",
+      }}
+    >
+      {new Array(20).fill("").map((ele, i) => (
```
Comment (Febinvarghese2) [inline src/components/HomeSkeleton.jsx]: @arun-121 by default are we setting it to 20 elements????

File: src/components/HomeSkeleton.jsx [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+        <Flex
+          key={i}
+          css={{
+            width: "260px",
+            height: "330px",
+            background: "lightgray",
+          }}
+        ></Flex>
```
Comment (Febinvarghese2) [inline src/components/HomeSkeleton.jsx]: @arun-121 btw this can be self closing tag right?

File: src/components/OrderSummary.jsx [hunk 30]
```diff
@@ -0,0 +1,80 @@
...
+              value: 0.2,
+            },
+            {
+              label: "30%",
+              value: 0.3,
+            },
+          ]}
+        />
+      </Flex>
+      <Flex css={{ border: "solid 1px gray" }}></Flex>
```
Comment (Febinvarghese2) [inline src/components/OrderSummary.jsx]: @arun-121 this can be `borderWidths$xs solid $neutral200`

File: src/components/OrderSummary.jsx [hunk 31]
```diff
@@ -0,0 +1,80 @@
...
+            totalAmount -
+            discount * (0.18 * totalAmount + totalAmount)
+          ).toFixed(3)}
+        </Text>
+      </Flex>
+      <Flex css={{ border: "solid 1px gray" }}></Flex>
+      <Button
+        size="md"
+        onClick={() =>
+          address === "" || address === null
```
Comment (Febinvarghese2) [inline src/components/OrderSummary.jsx]: @arun-121 this can be `!address`

File: src/components/ProductDetailView.jsx [hunk 32]
```diff
@@ -0,0 +1,78 @@
...
+            </Flex>
+            <Text weight={"bold"} size={"lg"}>
+              ₹ {selectedProduct?.price}
+            </Text>
+            <Text css={{ heigh: "100%" }}>{selectedProduct?.description}</Text>
+          </Flex>
+
+          <Button
+            onClick={() => dispatch(addToCart(selectedProduct?.id))}
+            css={{ height: "60px" }}
```
Comment (Febinvarghese2) [inline src/components/ProductDetailView.jsx]: @arun-121 you can use our theme sizes

File: src/components/ProductDetailView.jsx [hunk 33]
```diff
@@ -0,0 +1,78 @@
...
+          >
+            Add to Cart
+          </Button>
+        </Flex>
+      </Flex>
+      <Flex css={{ borderBottom: "solid 1px darkgray " }}> </Flex>
+      review
+      {/* <Text css={{ heigh: "100%" }}>{selectedProduct?.description}</Text> */}
```
Comment (Febinvarghese2) [inline src/components/ProductDetailView.jsx]: @arun-121 can you remove this commented code

File: src/components/ProductList.jsx [hunk 34]
```diff
@@ -0,0 +1,115 @@
...
+        css={{
+          flexWrap: "wrap",
+          gap: "$10",
+          overflow: "scroll",
+          padding: "$2",
+          width: "100%",
+        }}
+      >
+        {(filteredProducts.length == 0 ? products : filteredProducts).map(
+          (ele) => (
```
Comment (Febinvarghese2) [inline src/components/ProductList.jsx]: @arun-121 this can be named as product

File: src/components/SearchBar.jsx [hunk 35]
```diff
@@ -0,0 +1,98 @@
...
+const SearchBar = () => {
+  console.log("rendered");
+  const navigate = useNavigate();
+  const [query, setQuery] = useState("");
+  const [isSearchFocused, setSearchFocused] = useState(false);
+  const [suggestionUI, setSuggestionUI] = useState([]);
+  const [cachedResponse, setCachedResponse] = useState({});
+
+  function fetchSuggestions() {
+    console.log("rendered");
```
Comment (Febinvarghese2) [inline src/components/SearchBar.jsx]: @arun-121 can you remove all the console logs in the PR

File: src/components/SearchBar.jsx [hunk 36]
```diff
@@ -0,0 +1,98 @@
...
+        }}
+      >
+        <Input
+          onChange={(e) => setQuery(e.target.value)}
+          type="text"
+          onFocus={() => setSearchFocused(true)}
+          onBlur={() => setTimeout(() => setSearchFocused(false), 200)}
+          css={{
+            width: "500px",
+            height: "50px",
```
Comment (Febinvarghese2) [inline src/components/SearchBar.jsx]: @arun-121 can you use size prop instead of height?

File: src/components/SearchBar.jsx [hunk 37]
```diff
@@ -0,0 +1,98 @@
...
+            padding: "$2",
+            marginBottom: "$2",
+          }}
+          placeholder="search here"
```
Comment (Febinvarghese2) [inline src/components/SearchBar.jsx]: @arun-121 just Search... would be relevant right?

File: src/components/SideBar.jsx [hunk 38]
```diff
@@ -0,0 +1,54 @@
...
+        <Flex>
+          <img
+            src="https://asset.brandfetch.io/id4QM--uk4/idZgawl2X6.png"
+            width={"50px"}
+            height={"50px"}
+            alt=""
+          />
+        </Flex>
+        <Flex flexDirection="column" css={{ gap: "$20" }}>
+          <Flex>
```
Comment (Febinvarghese2) [inline src/components/SideBar.jsx]: @arun-121 is this flex needed?

File: src/components/SideBar.jsx [hunk 39]
```diff
@@ -0,0 +1,54 @@
...
+            <IoHome
+              onClick={() => navigate("/home")}
+              className="icon-hover"
```
Comment (Febinvarghese2) [inline src/components/SideBar.jsx]: @arun-121 is this class name needed if you want to achieve hover functionality in twigs you can do like this
```
<Flex css={{
"&:hover": { background: 'red' }
}}

File: src/pages/CartPage.jsx [hunk 40]
```diff
@@ -0,0 +1,21 @@
+import { Flex } from "@sparrowengg/twigs-react";
+import SideBar from "../components/SideBar";
+import CartItems from "../components/CartItems";
+import BagItems from "../components/BagItems";
+import { useSelector } from "react-redux";
+const CartPage = () => {
+  const cart = useSelector((state) => state.cart);
+  return (
+    <>
```
Comment (Febinvarghese2) [inline src/pages/CartPage.jsx:9]: @arun-121 can you remove unused react fragements

File: src/pages/CartPage.jsx [hunk 41]
```diff
@@ -0,0 +1,21 @@
...
+      <Flex css={{ height: "100vh" }}>
+        <SideBar />
+        <Flex flexDirection="column" css={{ width: "100%" }}>
+          <CartItems />
+        </Flex>
+        <Flex>{cart.length !== 0 && <BagItems />}</Flex>
```
Comment (Febinvarghese2) [inline src/pages/CartPage.jsx]: @arun-121 is this flex needed

File: src/pages/CheckOutPage.jsx [hunk 42]
```diff
@@ -0,0 +1,31 @@
...
+import BagItems from "../components/BagItems";
+import SideBar from "../components/SideBar";
+
+import { CheckoutDialog } from "../components/CheckoutDialog";
+import { useSelector } from "react-redux";
+import EmptyCart from "../components/EmptyCart";
+const CheckOutPage = () => {
+  const cart = useSelector((state) => state.cart);
+  return (
+    <Flex css={{ height: "100%", gap: "30px" }}>
```
Comment (Febinvarghese2) [inline src/pages/CheckOutPage.jsx]: @arun-121 this can be gap="$15"

File: src/pages/ProductPage.jsx [hunk 43]
```diff
@@ -0,0 +1,23 @@
...
+import ProductDetailView from "../components/ProductDetailView";
+import SideBar from "../components/SideBar";
+import BagItems from "../components/BagItems";
+const ProductPage = () => {
+  return (
+    <>
+      <Flex css={{ height: "100%" }} gap="$5">
+        <Flex>
+          <SideBar />
+        </Flex>
```
Comment (Febinvarghese2) [inline src/pages/ProductPage.jsx]: @arun-121 can you remove unused flexes here

File: src/redux/reducers/index.js [hunk 44]
```diff
@@ -0,0 +1,168 @@
...
+    priceUnder: [],
+    sortOrder: null,
+  },
+  address: null,
+  theme: "light",
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case "ADD_TO_CART": {
```
Comment (Febinvarghese2) [inline src/redux/reducers/index.js:17]: @arun-121 can you move these to constants so that the same values can be used in actions and in reducers

File: src/redux/reducers/index.js [hunk 45]
```diff
@@ -0,0 +1,168 @@
...
+      const itemId = parseInt(action.id);
+      const itemInCart = state.cart.find((ele) => ele.id === itemId);
+
+      if (itemInCart) {
+        const updatedCart = state.cart.map((ele) =>
+          ele.id === itemId ? { ...ele, cartCount: ele.cartCount + 1 } : ele
```
Comment (Febinvarghese2) [inline src/redux/reducers/index.js]: @arun-121 ca you give a relevant name for this

**Review by Febinvarghese2 — CHANGES_REQUESTED** [review]
@arun-121 can you check these comments

File: src/components/BagItems.jsx (hunk 13 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed to theming values

File: src/components/BagItems.jsx (hunk 12 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed to scroll auto

Comment (arun-121) [reply src/components/BagItems.jsx]: changed to wrap prop

File: src/components/BagItems.jsx (hunk 14 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed the same

File: src/components/BagItems.jsx (hunk 15 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed to theming values

File: src/components/BagItems.jsx (hunk 16 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed as product

File: src/components/CartItems.jsx (hunk 18 above)
Comment (arun-121) [reply src/components/CartItems.jsx]: removed comments

File: src/components/CartItems.jsx (hunk 19 above)
Comment (arun-121) [reply src/components/CartItems.jsx]: removed fontsize and used Heading component

File: src/components/CartItems.jsx (hunk 20 above)
Comment (arun-121) [reply src/components/CartItems.jsx:37]: changed to theming values

File: src/components/CartItems.jsx (hunk 21 above)
Comment (arun-121) [reply src/components/CartItems.jsx]: changed to template literal

File: src/components/CartItems.jsx (hunk 22 above)
Comment (arun-121) [reply src/components/CartItems.jsx]: changed to button component

File: src/components/CheckoutDialog.jsx (hunk 23 above)
Comment (arun-121) [reply src/components/CheckoutDialog.jsx]: removed the same

File: src/components/CheckoutDialog.jsx (hunk 24 above)
Comment (arun-121) [reply src/components/CheckoutDialog.jsx]: removed unnecessary wrapping

File: src/components/EmptyCart.jsx (hunk 25 above)
Comment (arun-121) [reply src/components/EmptyCart.jsx:4]: added space

File: src/components/EmptyCart.jsx (hunk 26 above)
Comment (arun-121) [reply src/components/EmptyCart.jsx]: added alt text

File: src/components/FilterMenu.jsx (hunk 27 above)
Comment (arun-121) [reply src/components/FilterMenu.jsx]: created a separate folder for constants

File: src/components/HomeSkeleton.jsx (hunk 29 above)
Comment (arun-121) [reply src/components/HomeSkeleton.jsx]: changed as self closing tag

File: src/components/OrderSummary.jsx (hunk 30 above)
Comment (arun-121) [reply src/components/OrderSummary.jsx]: changed as mentioned

File: src/components/ProductDetailView.jsx (hunk 32 above)
Comment (arun-121) [reply src/components/ProductDetailView.jsx]: changed to theme variables

File: src/components/ProductDetailView.jsx (hunk 33 above)
Comment (arun-121) [reply src/components/ProductDetailView.jsx]: removed comments

File: src/components/BagItems.jsx (hunk 17 above)
Comment (arun-121) [reply src/components/BagItems.jsx]: changed as !cart.length

File: src/components/ProductList.jsx (hunk 34 above)
Comment (arun-121) [reply src/components/ProductList.jsx]: changed as product

File: src/components/SearchBar.jsx (hunk 35 above)
Comment (arun-121) [reply src/components/SearchBar.jsx]: removed all console logs

File: src/components/SearchBar.jsx (hunk 36 above)
Comment (arun-121) [reply src/components/SearchBar.jsx]: used size prop

File: src/components/SearchBar.jsx (hunk 37 above)
Comment (arun-121) [reply src/components/SearchBar.jsx]: changed placeholder

File: src/components/SideBar.jsx (hunk 38 above)
Comment (arun-121) [reply src/components/SideBar.jsx]: removed unused flex's

File: src/components/SideBar.jsx (hunk 39 above)
Comment (arun-121) [reply src/components/SideBar.jsx]: removed classname

File: src/pages/CartPage.jsx (hunk 41 above)
Comment (arun-121) [reply src/pages/CartPage.jsx]: removed unnecessary flex

File: src/pages/ProductPage.jsx (hunk 43 above)
Comment (arun-121) [reply src/pages/ProductPage.jsx]: removed the same

File: src/redux/reducers/index.js (hunk 45 above)
Comment (arun-121) [reply src/redux/reducers/index.js]: renamed as cartItem

File: src/redux/reducers/index.js (hunk 44 above)
Comment (arun-121) [reply src/redux/reducers/index.js:17]: moved constants to a separate folder

File: src/pages/CheckOutPage.jsx (hunk 42 above)
Comment (arun-121) [reply src/pages/CheckOutPage.jsx]: changed as mentioned

File: src/pages/CartPage.jsx (hunk 40 above)
Comment (arun-121) [reply src/pages/CartPage.jsx:9]: removed unused fragments

File: src/components/OrderSummary.jsx (hunk 31 above)
Comment (arun-121) [reply src/components/OrderSummary.jsx]: made it as !address.trim() to ensure empty strings are not accepted

File: src/components/HomeSkeleton.jsx (hunk 28 above)
Comment (arun-121) [reply src/components/HomeSkeleton.jsx]: changed to  a constant variable instead of hardcoding
